### PR TITLE
feature/jenkins 36060 - personalization plugin

### PR DESCRIPTION
**Pull request by cliffmeyers**
Related to issue # JENKINS-36060. 

Summary of this pull request: 
* Create a new Maven module for the personalization plugin
* Demonstrate very simple POC of adding an item action to pipelines list. While testing this I uncovered an interesting bug in js-modules that Keith addressed in cloudbees/blueocean-plugin#303

@reviewbybees

### CODE REVIEW

File: blueocean-personalization/.storybook/config.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { configure } from '@kadira/storybook';
```
Comment (kzantow) [inline blueocean-personalization/.storybook/config.js]: Could all of this somehow be automated with js-modules or another devDependency?

File: blueocean-personalization/src/main/js/components/ActionLink.jsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import React, { Component } from 'react';
+
+export default class ActionLink extends Component {
+    render() {
+        return <div>Test</div>;
```
Comment (kzantow) [inline blueocean-personalization/src/main/js/components/ActionLink.jsx]: This part isn't actually ready for review, is it?

File: blueocean-personalization/src/main/resources/index.jelly [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<?jelly escape-by-default='true'?>
```
Comment (kzantow) [inline blueocean-personalization/src/main/resources/index.jelly]: What's this file for?

File: blueocean-plugin/pom.xml [hunk 4]
```diff
@@ -24,6 +24,10 @@
         </dependency>
         <dependency>
             <groupId>${project.groupId}</groupId>
+            <artifactId>blueocean-personalization</artifactId>
```
Comment (kzantow) [inline blueocean-plugin/pom.xml]: Need to add this to <modules> section, too?

File: blueocean-plugin/src/test/java/io/jenkins/blueocean/jsextensions/JenkinsJSExtensionsTest.java [hunk 5]
```diff
@@ -40,18 +39,28 @@
     @Test
     public void test() {
         // Simple test of the rest endpoint. It should find the "blueocean-dashboard"
```
Comment (kzantow) [inline blueocean-plugin/src/test/java/io/jenkins/blueocean/jsextensions/JenkinsJSExtensionsTest.java]: NIT: 'blueocean-dashboard'

File: blueocean-plugin/pom.xml (hunk 4 above)
Comment (cliffmeyers) [inline blueocean-plugin/pom.xml]: Yes, it's added to `<modules>` as well.

Comment (kzantow) [inline blueocean-plugin/pom.xml]: Ah, whoops, wrong pom :)

**Comment by kzantow** [comment]
LGTM except the `ActionLink.jsx` for now

File: blueocean-personalization/src/main/js/components/ActionLink.jsx (hunk 2 above)
Comment (cliffmeyers) [inline blueocean-personalization/src/main/js/components/ActionLink.jsx]: You're right, not good to litter Test text all over the main dashboard, even for a day or two until the other tickets get completed. For now I updated it just to write a HTML comment in there so I can still easily prove / test the integration (see commit 01ffc01). Will that work?

**Comment by kzantow** [comment]
:bee: from me, nice job

**Comment by michaelneale** [comment]
can we defer merging this until multibranch is fixed? https://issues.jenkins-ci.org/browse/JENKINS-36338

File: blueocean-personalization/src/main/js/components/ActionLink.jsx [hunk 6]
```diff
@@ -2,6 +2,6 @@ import React, { Component } from 'react';
 
 export default class ActionLink extends Component {
     render() {
-        return <div>Test</div>;
+        return <div dangerouslySetInnerHTML={{ __html: '<!--Test-->' }}></div>;
```
Comment (michaelneale) [inline blueocean-personalization/src/main/js/components/ActionLink.jsx]: love that method name from react ;)

File: blueocean-personalization/src/main/resources/index.jelly (hunk 3 above)
Comment (michaelneale) [inline blueocean-personalization/src/main/resources/index.jelly]: every plugin I think needs an index.jelly - I think...

**Comment by michaelneale** [comment]
🐝  as code seems ok and keith seems happy, but not sure what am looking for (story book didn't show up anything for me)

**Comment by cliffmeyers** [comment]
This is really just a shell of the plugin, since I have other JIRA tickets for the meaningful feature delivery. Still nice to keep this in a separate PR so we can get an idea of what needs to be added for new plugins. Going to merge now and will have a PR later today with some storybook for the Pipeline Card component.

**Pull request merged**